### ISSUE TITLE
binary (pdf) tests doesn't work: engine specific pdf is missing

### ISSUE THREAD

**Issue by u-fischer** [latex3/l3build#62]
my build.lua looks like this

~~~~
-- Build script for test
module   = "buildbug"

checkengines = {"pdftex"}


if not release_date then
 kpse.set_program_name ("kpsewhich")
 dofile ( kpse.lookup ("l3build.lua"))
end
~~~~

In the testfiles folder I have this test.lvt

~~~~
\input{regression-test}
\documentclass{article}

\START
\begin{document}
some text
\end{document}

~~~~

Running `l3build save test` and then `l3build check` works fine, but 
`l3build save -p test` ends with the message `Das System kann die angegebene Datei nicht finden.` (file no found), and a following check fails as TEST.PDFTEX.PDF is not found:

~~~~
G:\buildbug>l3build check
Running checks on
  test (1/1)
FC: Kann ./BUILD/TEST/TEST.PDFTEX.PDF nicht öffnen - Datei oder Ordner nicht vorhanden

          --> failed

  Check failed with difference files
  - ./build/test/test.pdftex.cmp
~~~~
     
The testfiles folder contains only a test.pdf.  The `build/test` folder contains after the save step a `test.pdf` and after the check step also a `test.ref.pdf`.

**Comment by josephwright** [comment]
Hmm, I'm wondering whether to fix this now or to do the proper rework ...

**Comment by josephwright** [comment]
I think I'll take advantage of the long weekend to look again at PDF-based testing entirely: hopefully that will address the issue.

**Comment by josephwright** [comment]
This one should be fixed, though see comments in #10 and #21 (and change of binary test set up).
